official/projects/detr/tasks/detection.py

```diff
@@ -37,46 +44,106 @@ class DectectionTask(base_task.Task):
 
   def build_model(self):
     """Build DETR model."""
+
+    input_specs = tf.keras.layers.InputSpec(
+        shape=[None] + self._task_config.model.input_size)
+
+    backbone = backbones.factory.build_backbone(
+        input_specs=input_specs,
+        backbone_config=self._task_config.model.backbone,
+        norm_activation_config=self._task_config.model.norm_activation)
     model = detr.DETR(
-        self._task_config.num_queries,
-        self._task_config.num_hidden,
-        self._task_config.num_classes,
-        self._task_config.num_encoder_layers,
-        self._task_config.num_decoder_layers)
+        backbone,
+        self._task_config.model.num_queries,
+        self._task_config.model.hidden_size,
+        self._task_config.model.num_classes,
+        self._task_config.model.num_encoder_layers,
+        self._task_config.model.num_decoder_layers)
     return model
 
   def initialize(self, model: tf.keras.Model):
     """Loading pretrained checkpoint."""
-    ckpt = tf.train.Checkpoint(backbone=model.backbone)
-    status = ckpt.read(self._task_config.init_ckpt)
-    status.expect_partial().assert_existing_objects_matched()
-
-  def build_inputs(self, params, input_context=None):
+    if not self._task_config.init_checkpoint:
+      return
+
+    ckpt_dir_or_file = self._task_config.init_checkpoint
+
+    # Restoring checkpoint.
+    if tf.io.gfile.isdir(ckpt_dir_or_file):
+      ckpt_dir_or_file = tf.train.latest_checkpoint(ckpt_dir_or_file)
+
+    if self._task_config.init_checkpoint_modules == 'all':
+      ckpt = tf.train.Checkpoint(**model.checkpoint_items)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.assert_consumed()
+    elif self._task_config.init_checkpoint_modules == 'backbone':
+      ckpt = tf.train.Checkpoint(backbone=model.backbone)
+      status = ckpt.restore(ckpt_dir_or_file)
+      status.expect_partial().assert_existing_objects_matched()
+
+    logging.info('Finished loading pretrained checkpoint from %s',
+                 ckpt_dir_or_file)
+
+  """def build_inputs(self,
+                   params: detr_cfg.DataConfig,
+                   input_context: Optional[tf.distribute.InputContext] = None):
+    return coco.COCODataLoader(params).load(input_context)"""
+  
+  def build_inputs(self,
+                   params,
+                   input_context: Optional[tf.distribute.InputContext] = None):
     """Build input dataset."""
-    return coco.COCODataLoader(params).load(input_context)
```

Comment:
should we add a if statement
if params is COCODataConfig:
  return coco.COCODataLoader(params).load(input_context)

so we keep both path working and deprecate it later.